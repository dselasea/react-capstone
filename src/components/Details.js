import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink, useParams } from 'react-router-dom';
import { filterCrypto } from '../redux/data/dataSlice';

const Details = () => {
  const cryptoInfo = useSelector((state) => state.data);
  const dispatch = useDispatch();

  const { id } = useParams();

  useEffect(() => {
    dispatch(filterCrypto(id));
  }, [dispatch, id]);

  const crypto = cryptoInfo.data.filter((cryptoData) => cryptoData.show === true);

  return (
    <>
      <NavLink to="/">
        <button type="button" className="btn">Back</button>
      </NavLink>
      <div className="crypto-detail">
        {
        crypto.map((cryptoData) => (
          <div key={cryptoData.id}>
            <img src={cryptoData.cryptoImage} alt={cryptoData.cryptoName} />
            <h1>
              Rank:
              {' '}
              {cryptoData.cryptoRank}
            </h1>
            <h2>{cryptoData.cryptoSymbol}</h2>

            <div className="details-card">
              <h4>Price:</h4>
              {' '}
              $
              {cryptoData.cryptoPrice.toFixed(2)}
            </div>
            <div className="details-card">
              <h4>Website:</h4>
              {cryptoData.cryptoUrl}
            </div>
            <div className="details-card">
              <h4>Price Change (Hour):</h4>
              {cryptoData.cryptoChangeHour}
            </div>
            <div className="details-card">
              <h4>Price Change (Day):</h4>
              {cryptoData.cryptoChangeDay}
            </div>
            <div className="details-card">
              <h4>Price Change (Week):</h4>
              {cryptoData.cryptoChangeWeek}
            </div>
          </div>
        ))
      }
      </div>
    </>
  );
};

export default Details;
