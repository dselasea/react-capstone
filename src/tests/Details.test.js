import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Details from '../components/Details';

describe('Home', () => {
  it('renders correctly', () => {
    const detailsComponent = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Details />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(detailsComponent).toMatchSnapshot();
  });
});
