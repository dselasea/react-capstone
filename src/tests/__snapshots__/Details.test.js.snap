// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Array [
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <button
      className="btn"
      type="button"
    >
      Back
    </button>
  </a>,
  <section
    className="crypto-detail"
  />,
]
`;
